src/SortedSet/ArrayStrategy.coffee

```diff
@@ -35,15 +36,18 @@ binarySearchForIndex = (array, value, comparator) ->
 
 class ArrayStrategy
   constructor: (@options) ->
+    @onInsertConflict = @options.onInsertConflict
     @comparator = @options.comparator
     @data = []
 
   toArray: -> @data
 
   insert: (value) ->
     index = binarySearchForIndex(@data, value, @comparator)
-    throw 'Value already in set' if @data[index] == value
-    @data.splice(index, 0, value)
+    if @data[index] != undefined and @comparator(@data[index], value) == 0
```

Comment:
Why change the if-statement condition? (It may be correct; but please write a test that fails without this change.)

Reply:
Strict equality comparison doesn't work when, for example, one is working with different objects having identical shapes. I will double-check but I am under the impression that the current tests would already fail without this. I haven't really verified this, though. Thank you for reminding me.

Reply:
I've checked - the new tests do indeed fail without changing the `if` condition. The same happens for the `RedBlackTreeStrategy`. I would consider this resolved but the final say is yours.